test/patch/reverse.js

```diff
@@ -0,0 +1,97 @@
+        'index 20b807a..4a96aff 100644\n' +
+        '--- a/CONTRIBUTING.md\n' +
+        '+++ b/CONTRIBUTING.md\n' +
+        '@@ -2,6 +2,8 @@\n' +
+        ' \n' +
+        ' ## Pull Requests\n' +
+        ' \n' +
+        '+bla bla bla\n' +
+        '+\n' +
+        ' We also accept [pull requests][pull-request]!\n' +
+        ' \n' +
+        ' Generally we like to see pull requests that\n' +
+        'diff --git a/README.md b/README.md\n' +
+        'index 06eebfa..40919a6 100644\n' +
+        '--- a/README.md\n' +
+        '+++ b/README.md\n' +
+        '@@ -1,5 +1,7 @@\n' +
+        ' # jsdiff\n' +
+        ' \n' +
+        '+foo\n' +
+        '+\n' +
+        ' [![Build Status](https://secure.travis-ci.org/kpdecker/jsdiff.svg)](http://travis-ci.org/kpdecker/jsdiff)\n' +
+        ' [![Sauce Test Status](https://saucelabs.com/buildstatus/jsdiff)](https://saucelabs.com/u/jsdiff)\n' +
+        ' \n' +
+        "@@ -225,3 +227,5 @@ jsdiff deviates from the published algorithm in a couple of ways that don't affe\n" +
+        ' \n' +
+        " * jsdiff keeps track of the diff for each diagonal using a linked list of change objects for each diagonal, rather than the historical array of furthest-reaching D-paths on each diagonal contemplated on page 8 of Myers's paper.\n" +
+        ' * jsdiff skips considering diagonals where the furthest-reaching D-path would go off the edge of the edit graph. This dramatically reduces the time cost (from quadratic to linear) in cases where the new text just appends or truncates content at the end of the old text.\n' +
+        '+\n' +
+        '+bar\n'
+      );
+      expect(formatPatch(reversePatch(patch))).to.equal(
+        '===================================================================\n' +
+        '--- b/README.md\t\n' +
+        '+++ a/README.md\t\n' +
+        '@@ -1,7 +1,5 @@\n' +
+        ' # jsdiff\n' +
+        ' \n' +
+        '-foo\n' +
+        '-\n' +
+        ' [![Build Status](https://secure.travis-ci.org/kpdecker/jsdiff.svg)](http://travis-ci.org/kpdecker/jsdiff)\n' +
+        ' [![Sauce Test Status](https://saucelabs.com/buildstatus/jsdiff)](https://saucelabs.com/u/jsdiff)\n' +
+        ' \n' +
+        '@@ -227,5 +225,3 @@\n' +
+        ' \n' +
+        " * jsdiff keeps track of the diff for each diagonal using a linked list of change objects for each diagonal, rather than the historical array of furthest-reaching D-paths on each diagonal contemplated on page 8 of Myers's paper.\n" +
+        ' * jsdiff skips considering diagonals where the furthest-reaching D-path would go off the edge of the edit graph. This dramatically reduces the time cost (from quadratic to linear) in cases where the new text just appends or truncates content at the end of the old text.\n' +
+        '-\n' +
+        '-bar\n' +
+        '\n' +
+        '===================================================================\n' +
+        '--- b/CONTRIBUTING.md\t\n' +
+        '+++ a/CONTRIBUTING.md\t\n' +
```

Comment:
I think it's a pre-existing issue orthogonal to the main point of this PR, but the way the original header gets screwed up here is a bit distasteful. I should dig in and make an issue before merging this.

Reply:
Issue: https://github.com/kpdecker/jsdiff/issues/454